### PR TITLE
feat: HIP-904: add Token Cancel Airdrop proto

### PR DESCRIPTION
**Pull request by stoyanov-st**
**Description**:
Added protobuf records for TokenCancelAirdrop transaction

**Related issue(s)**:
[13997](https://github.com/hashgraph/hedera-services/issues/13997)
Fixes #

**Notes for reviewer**:

❗ We don't want to merge this before https://github.com/hashgraph/hedera-services/pull/13830 is ready and merged

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: services/token_cancel_airdrop.proto [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+option java_package = "com.hederahashgraph.api.proto.java";
+// <<<pbj.java_package = "com.hedera.hapi.node.token">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+import "basic_types.proto";
+
+/**
+ * Token cancel airdrop<br/>
+ * Remove one or more pending airdrops from state on behalf of the sender(s)
+ * for each airdrop.<br/>
```
Comment (jsync-swirlds) [inline services/token_cancel_airdrop.proto:43]: `<br/>` is not needed at end of this line.

**Review by jsync-swirlds — COMMENTED** [review]
Looks good, just a few suggested consistency improvements to specification text.

File: services/token_cancel_airdrop.proto [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ */
+message TokenCancelAirdropTransactionBody {
+  /**
+   * A list of one or more pending airdrop identifiers.<br/>
+   * This transaction MUST be signed by the account referenced by a `sender_id` for
+   * each entry in this list.
+   * <p>
+   * This list MUST contain between 1 and 10 entries, inclusive.
+   * This list MUST NOT have any duplicate entries.<br/>
+   */
```
[the remark this answers is not in the document]
Comment (stoyanov-st) [reply services/token_cancel_airdrop.proto]: Thank you! I've added the redacted version.

File: services/token_cancel_airdrop.proto [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119).
+ */
+syntax = "proto3";
+
+package proto;
+
+/*-
+ *
+ * Hedera Network Services Protobuf
+ *
```
[the remark this answers is not in the document]
Comment (stoyanov-st) [reply services/token_cancel_airdrop.proto]: Removed it. Thanks!

File: services/token_cancel_airdrop.proto [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+import "basic_types.proto";
+
+/**
+ * Token cancel airdrop<br/>
+ * Remove one or more pending airdrops from state on behalf of the sender(s)
+ * for each airdrop.<br/>
+ *
+ * Each pending airdrop canceled SHALL be removed from state and SHALL NOT be available to claim.
+ *
```
[the remark this answers is not in the document]
Comment (stoyanov-st) [reply services/token_cancel_airdrop.proto]: Understood. Line break added.

File: services/token_cancel_airdrop.proto [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+ * Each cancellation SHALL be represented in the transaction body and SHALL NOT be restated
+ * in the record file.
```
[the remark this answers is not in the document]
Comment (stoyanov-st) [reply services/token_cancel_airdrop.proto]: Line break added.

File: services/token_service.proto [hunk 6]
```diff
@@ -174,4 +174,10 @@ service TokenService {
      * tokens, SHALL be charged to the account submitting this transaction.
      */
     rpc airdropTokens (Transaction) returns (TransactionResponse);
+
+    /**
+     * Cancel one or more pending airdrops.<br/>
+     * This transaction MUST be signed by _each_ account *sending* an airdrop to be canceled.
```
[the remark this answers is not in the document]
Comment (stoyanov-st) [reply services/token_service.proto]: Done.

File: services/token_cancel_airdrop.proto (hunk 1 above)
Comment (stoyanov-st) [reply services/token_cancel_airdrop.proto]: Removed it. Thanks!

**Review by jsync-swirlds — APPROVED** [review]

**Pull request merged**